File: Gymnasium/DummyEnv.cs
using System;
using System.Collections.Generic;
using Gymnasium.Spaces;

namespace Gymnasium;

/// <summary>
/// DummyEnv: simple demonstration environment for testing.
/// </summary>
public class DummyEnv : Env<int, int>
{
    private int _state;

    public Discrete ActionSpace { get; } = new Discrete(3); // Actions: 0, 1, 2
    public Discrete ObservationSpace { get; } = new Discrete(20); // States: 0-19

    public override int Reset()
    {
        _state = 0;
        return _state;
    }

    public override (int state, double reward, bool done, IDictionary<string, object> info) Step(int action)
    {
        _state += action;
        double reward = _state;
        bool done = _state >= 10;
        var info = new Dictionary<string, object>();
        return (_state, reward, done, info);
    }

    public override void Render(string mode = "human")
    {
        Console.WriteLine($"State: {_state}");
    }

    public override void Close() { }
}

File: Gym/dummy_env.py
import numpy as np
from .env import Env
from typing import Any, Tuple

class DummyEnv(Env):
    """A simple dummy environment for demonstration purposes."""
    def __init__(self):
        self.state = 0

    def reset(self) -> int:
        self.state = 0
        return self.state

    def step(self, action: int) -> Tuple[int, float, bool, dict]:
        self.state += action
        reward = float(self.state)
        done = self.state >= 10
        info = {}
        return self.state, reward, done, info

    def render(self, mode: str = "human") -> None:
        print(f"State: {self.state}")

    def close(self) -> None:
        pass
